packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-gssapi.tsx

```diff
@@ -30,13 +44,22 @@ function AuthenticationGSSAPI({
     'kerberosPrincipal'
   );
   const principal = getConnectionStringUsername(connectionStringUrl);
+  const password = getConnectionStringPassword(connectionStringUrl);
+
   const authMechanismProperties =
     parseAuthMechanismProperties(connectionStringUrl);
   const serviceName = authMechanismProperties.get('SERVICE_NAME');
   const serviceRealm = authMechanismProperties.get('SERVICE_REALM');
-  const canonicalizeHostname = authMechanismProperties.get(
-    'CANONICALIZE_HOST_NAME'
-  );
+  const canonicalizeHostname =
+    authMechanismProperties.get('CANONICALIZE_HOST_NAME') || 'none';
+
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!showPassword && password.length) {
+      setShowPassword(true);
+    }
+  }, [password, showPassword, updateConnectionFormField]);
```

Comment:
If I'm understanding the logic right, I think this effect tries to check too much, the only thing it should care about is if password exists or not, right? And then the state change of the checkbox should be the trigger for resetting the password, something like this maybe?

```suggestion
  // If connection string password changed and there is one now, check the checkbox automatically
  useEffect(() => {
    if (password.length) {
      setShowPassword(true);
    }
  }, [password]);
  
  // When password field is hidden, clean the password from the connection string
  useEffect(() => {
    if (!showPassword) {
      updateConnectionFormField({
        type: 'update-password',
        password: '',
      });
    }    
  }, [showPassword, updateConnectionFormField])
```

This also means that you don't need to handle `updateConnectionFormField` in multiple places and the checkbox input only need to change the state of the `showPassword` and nothing else

Reply:
That's nicer yeah, won't that need to wait for the re-render in order to clean the password? or `useEffect` is completely detached?

Reply:
It would need to, yeah (the flow would be something like this: uncheck input -> state change [showPassword is false now] -> render -> effect [because showPassword changed] -> reducer change), but it would be absolutely instant in this case, I wouldn't be concerned about this. But also I submitted the review before I saw that it got approved and merged already and it's alright like it is now too